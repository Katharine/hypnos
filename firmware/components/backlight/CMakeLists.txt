idf_component_register(SRCS "backlight.cpp"
                    INCLUDE_DIRS "include"
                    REQUIRES lvgl_port esp_adc
                    PRIV_REQUIRES log)
