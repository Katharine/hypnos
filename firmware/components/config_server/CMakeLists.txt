idf_component_register(SRCS "config_server.cpp"
                    INCLUDE_DIRS "include"
                    REQUIRES hypnos_config wifi eightsleep
                    PRIV_REQUIRES esp_http_server arduinojson log)

component_compile_options("-Wno-missing-field-initializers")

# Magic to include the config webpage.
include(CMakeRC.cmake)
cmrc_add_resource_library(web-contents NAMESPACE web WHENCE resources resources/index.html)
target_link_libraries(${COMPONENT_LIB} PRIVATE web-contents)
