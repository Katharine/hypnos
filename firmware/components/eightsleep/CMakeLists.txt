idf_component_register(SRCS "eightsleep.cpp" "internal/apiclient.cpp"
                    INCLUDE_DIRS "include"
                    REQUIRES http_cxx expected arduinojson)

component_compile_options("-Wno-missing-field-initializers")
