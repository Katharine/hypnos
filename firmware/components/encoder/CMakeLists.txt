idf_component_register(SRCS "encoder.cpp"
                    INCLUDE_DIRS "include"
                    REQUIRES driver log lvgl)
