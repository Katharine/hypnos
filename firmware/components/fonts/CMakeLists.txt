idf_component_register(SRCS "montserrat_56_digits.c"
                    INCLUDE_DIRS "include"
                    REQUIRES lvgl)
