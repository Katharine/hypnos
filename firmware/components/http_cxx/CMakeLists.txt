idf_component_register(SRCS "http_cxx.cpp"
                    INCLUDE_DIRS "include"
        REQUIRES esp_http_client
        PRIV_REQUIRES mbedtls log esp-tls)

component_compile_options("-Wno-missing-field-initializers")