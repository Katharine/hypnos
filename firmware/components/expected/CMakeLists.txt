idf_component_register(SRCS "expected.cpp"
                    INCLUDE_DIRS "include")
